crabml-core/src/backends/cpu/buf/buf_q8_0.rs

```diff
@@ -236,27 +236,29 @@ mod impl_x86_64_avx2 {
     /// Inspired a lot by ggml https://github.com/ggerganov/ggml/blob/master/src/ggml-quants.c
 
     pub fn quantize_f32_q8_0(data: &[f32]) -> Vec<BlockQ8_0> {
+        debug_assert_eq!(data.len() % 32, 0);
+
         let mut bs = Vec::with_capacity(data.len() / 32);
 
         unsafe {
+            let mask = _mm256_set1_ps(-0.0);
+
             for chunk in data.chunks(32) {
                 let mut max_abs_values = _mm256_setzero_ps();
 
-                for &value in chunk {
-                    let val_vec = _mm256_set1_ps(value);
-                    max_abs_values = _mm256_max_ps(
-                        max_abs_values,
-                        _mm256_andnot_ps(_mm256_set1_ps(-0.0), val_vec),
-                    );
+                for values in chunk.chunks(8) {
+                    let value_vec = _mm256_loadu_ps(values.as_ptr());
+                    max_abs_values =
+                        _mm256_max_ps(max_abs_values, _mm256_andnot_ps(mask, value_vec))
                 }
 
                 let max_abs_value = {
                     let mut max_vals = [0.0; 8];
                     _mm256_storeu_ps(max_vals.as_mut_ptr(), max_abs_values);
                     *max_vals
                         .iter()
-                        .max_by(|x, y| x.partial_cmp(y).unwrap())
-                        .unwrap()
+                        .max_by(|x, y| x.partial_cmp(y).unwrap_unchecked())
+                        .unwrap_unchecked()
                 };
 
                 let d = max_abs_value / 127.0;
```

Comment:
Hi @flaneur2020 . Just a reminder, please make sure the code here is modified correctly. There are a few similar code clips in the repo. If the optimization is correct, I can help modify all of them.

Reply:
thank you for the suggestion, IMHO using `unwrap_unchecked()` is a nice to have in these unsafe blocks, we're not afraid of shooting ourselves in these unsafe computations :)

Reply:
Emm, actually I mean the changes between L245-L250 (old) and L249-L252 (new).

Reply:
I thought the previous code didn't utilize the power of `simd`. But I'm not sure if I'm right.

Reply:
oh i see, let's do it